pkg/compression/compression.go

```diff
@@ -35,32 +36,85 @@ func XzDecompressor(r io.Reader) (io.ReadCloser, error) {
 	return ioutil.NopCloser(r), nil
 }
 
+// compressorFunc returns the compressed stream, given a compressed stream.
+// The caller must call Close() on the compressed stream (even if the compressed input stream does not need closing!).
+type compressorFunc func(io.Writer, int) (io.WriteCloser, error)
+
+// GzipCompressor is a CompressorFunc for the gzip compression algorithm.
+func GzipCompressor(r io.Writer, level int) (io.WriteCloser, error) {
+	if level > 0 {
```

Comment:
In `pgzip`, unlike `zstd`, AFAICS `0` is a valid level value, different from “use the built-in default”.

Either define a custom scale, or make the level a `*int` or something similar, to clearly express the distinction between the default and a user-specified value, regardless of the semantics of the algorithm.